content/en/examples/application/guestbook/frontend-deployment.yaml

```diff
@@ -1,38 +1,37 @@
-apiVersion: apps/v1 # for versions before 1.9.0 use apps/v1beta2
+apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: frontend
   labels:
-    app: guestbook
+    app.kubernetes.io/name: guestbook
+    app.kubernetes.io/component: frontend
 spec:
   selector:
     matchLabels:
-      app: guestbook
-      tier: frontend
+      app.kubernetes.io/name: guestbook
+      app.kubernetes.io/component: frontend
   replicas: 3
   template:
     metadata:
       labels:
-        app: guestbook
-        tier: frontend
+        app.kubernetes.io/name: guestbook
+        app.kubernetes.io/component: frontend
     spec:
       containers:
-      - name: php-redis
-        image: gcr.io/google-samples/gb-frontend:v4
+      - name: guestbook
+        image: paulczar/gb-frontend:v5
+        # image: gcr.io/google-samples/gb-frontend:v4
         resources:
           requests:
             cpu: 100m
             memory: 100Mi
         env:
         - name: GET_HOSTS_FROM
           value: dns
-          # Using `GET_HOSTS_FROM=dns` requires your cluster to
-          # provide a dns service. As of Kubernetes 1.3, DNS is a built-in
-          # service launched automatically. However, if the cluster you are using
-          # does not have a built-in DNS service, you can instead
-          # access an environment variable to find the master
-          # service's host. To do so, comment out the 'value: dns' line above, and
-          # uncomment the line below:
+          # If your cluster config does not include a dns service, then to
+          # instead access environment variables to find service host
```

Comment:
Suggest reword comment for readability,
If your cluster config does not include a DNS service, then you can
access environment variables to find service host info.
Comment out the `value: dns` line above and remove the comment
for the field, `value: env`.

Reply:
I feel like we could just remove that entirely ... do we really think people are using kube without kube-dns ?

Reply:
Yep. People are either using CoreDNS or kube-dns, and name resolution is reasonable to take for granted.